crates/core/tedge_agent/src/agent.rs

```diff
@@ -179,7 +180,8 @@ impl SmAgentConfig {
         let mqtt_config = mqtt_channel::Config::default()
             .with_host(tedge_config.query(MqttBindAddressSetting)?.to_string())
             .with_port(tedge_config.query(MqttPortSetting)?.into())
-            .with_max_packet_size(10 * 1024 * 1024);
+            .with_max_packet_size(10 * 1024 * 1024)
+            .with_last_will_message(get_health_status_down_message("tedge-agent"));
```

Comment:
Minor: The string literal "tedge-agent" is reused here and in the `send_health_status` function. Better to move it to a const.

Reply:
Good point. It's also strange that no `session_name` is set. The same const should be used for both.

Reply:
fixed